src/Microsoft.DotNet.Interactive.Http.Tests/HttpKernelTests.NamedRequest.cs

```diff
@@ -0,0 +1,1070 @@
+                        headerValue = values.First();
+                    }
+
+                    response.RequestMessage = message;
+                    var contentString = """
+             {
+                "headers": {
+                    "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7",
+                    "Accept-Encoding": "gzip, deflate, br, zstd",
+                    "Accept-Language": "en-US,en;q=0.9",
+                    "Host": "httpbin.org",
+                    "Priority": "u=0, i",
+                    "Sec-Ch-Ua": "\""Chromium\"";v=\""128\"", \""Not;A=Brand\"";v=\""24\"", \""Microsoft Edge\"";v=\""128\"",
+                    "Sec-Ch-Ua-Mobile": "?0",
+                    "Sec-Ch-Ua-Platform": "\""Windows\"",
+                    "Sec-Fetch-Dest": "document",
+                    "Sec-Fetch-Mode": "navigate",
+                    "Sec-Fetch-Site": "none",
+                    "Sec-Fetch-User": "?1",
+                    "Upgrade-Insecure-Requests": "1",
+                    "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36 Edg/128.0.0.0",
+                    "X-Amzn-Trace-Id": "Root=1-66e9f16e-7afea6f643e754f854c57d85"
+                }
+            }
+            """;
+                    response.Content = new StringContent(contentString, Encoding.UTF8, "application/json");
+                    response.Headers.Add("server", "gunicorn/19.9.0");
+                    return Task.FromResult(response);
+                });
+                var client = new HttpClient(responseHandler);
+                using var kernel = new HttpKernel("http", client);
+
+                var firstCode = """
+            @baseUrl = https://httpbin.org/headers
+
+            # @name binHeader
+            GET {{baseUrl}}
+
+            ###
+            """;
+
+                var firstResult = await kernel.SendAsync(new SubmitCode(firstCode));
+                firstResult.Events.Should().NotContainErrors();
+
+
+                var secondCode = $$$"""
+            GET https://httpbin.org/headers
+            Detectedserver: {{binHeader.response.headers.Accept}}
+            ###
+            """;
+
+                var secondResult = await kernel.SendAsync(new SubmitCode(secondCode));
+                var diagnostics = secondResult.Events.Should().ContainSingle<DiagnosticsProduced>().Which;
+                diagnostics.Diagnostics.First().Message.Should().Be($$$"""The supplied header name 'Accept' does not exist in the named request.""");
+            }
+
+            [Theory]
+            [InlineData("json.response.body.$.slideshow.slides.title", "Wake up to WonderWidgets!")]
+            [InlineData("json.response.body.$.slideshow.slides.type", "all")]
+            public async Task json_with_additional_syntax_depths_can_be_accessed_correctly(string path, string end)
```

Comment:
Nitpick: Renaming `end` to something more intention-revealing (e.g. `expectedValue`) would help the readability of the test.